moveit_experimental/moveit_jog_arm/src/jog_calcs.cpp

```diff
@@ -111,9 +108,28 @@ void JogCalcs::startMainLoop(JogArmShared& shared_variables, std::mutex& mutex)
   {
     ros::Duration(WHILE_LOOP_WAIT).sleep();
 
+    // Ensure the low-pass filter matches reality
+    for (std::size_t i = 0; i < num_joints_; ++i)
+      position_filters_[i].reset(joint_state_.position[i]);
+
+    //  Check for a new command
     mutex.lock();
     cartesian_deltas = shared_variables.command_deltas;
     joint_deltas = shared_variables.joint_command_deltas;
+    incoming_joints_ = shared_variables.joints;
+    mutex.unlock();
+
+    kinematic_state_->setVariableValues(joint_state_);
+
+    // Always update the end-effector transform in case the getCommandFrameTransform() method is being used
+    // Get the transform from MoveIt planning frame to jogging command frame
+    // We solve (planning_frame -> base -> robot_link_command_frame)
+    // by computing (base->planning_frame)^-1 * (base->robot_link_command_frame)
+    tf_moveit_to_cmd_frame_ = kinematic_state_->getGlobalLinkTransform(parameters_.planning_frame).inverse() *
```

Comment:
Does this really fix the issue or is it just much more unlikely to occur? To me it seems like there is still the possibility of a race condition when getCommandFrameTransform() is called before we have an updated joint state. Also, wouldn't it be better to compute `tf_moveit_to_cmd_frame` inside getCommandFrameTransform() in order to reduce redundant computations? I don't see the transform being used here anymore. Maybe we should additionally let `getCommandFrameTransform()` return bool or add a blocking function `waitForValidJointState()`.

Reply:
Line 94 ensures joint_state_ is updated before this, so no race condition.

Reply:
Yeah, I guess it does make sense to calculated `tf_moveit_to_cmd_frame` inside the getter method.

Reply:
However, I did notice that `tf_planning_to_cmd_frame` was redundant with `tf_moveit_to_cmd_frame_` and could be deleted

Reply:
I added an `isInitialized()` function and `std::atomic<bool> is_initialized_` like you added previously for `stop_requested_`

Reply:
I thought it would be enough to just access `shared_variables.joints` for getting the latest state. Anyway, this doesn't really belong into this PR and I'm fine with the current solution